src/kernels.jl

```diff
@@ -4,6 +4,22 @@ function get_array_type(a::AT) where AT <: AbstractArray
     return AT.name.wrapper
 end
 
+@inline function sum_pairwise_forces(inters, coord_i, coord_j, atom_i, atom_j,
+                                    boundary, special, ::Val{F}) where F
+    dr = vector(coord_i, coord_j, boundary)
+    f_tuple = ntuple(length(inters)) do inter_type_i
+        force_gpu(inters[inter_type_i], dr, coord_i, coord_j, atom_i, atom_j, boundary, special)
+    end
+    f = sum(f_tuple)
+    if unit(f[1]) != F
+        # This triggers an error but it isn't printed
+        # See https://discourse.julialang.org/t/error-handling-in-cuda-kernels/79692
+        #   for how to throw a more meaningful error
+        error("wrong force unit returned, was expecting $F but got $(unit(f[1]))")
```

Comment:
The interpolation here is particularly tricky. I would avoid that if at all possible.

Reply:
To be clear: you are referring to the `error(...)` call in an inlined function within an `@kernel`? Or carrying the units through to this stage in the first place?

Reply:
`error("Oops")` is fine, `error("Oops, $F")` is sadly not since it string interpolation is really tough on the GPU compiler.

Reply:
We can remove that, no problem.